File: leetcode/algorithms/array/average-salary-excluding-the-minimum-and-maximum-salary.js
/*
	leetcode
	category: "Algorithms"
	problem title: "Average Salary Excluding the Minimum and Maximum Salary"
*/


/**
 * @param {number[]} salary
 * @return {number}
 */
var average = function(salary) {
    return salary.sort((a,b) => a-b).reduce((avg,sal,index) => {
        if(index>0 && index<salary.length-1){
            avg = ((avg*(index-1))+sal)/index;
        }
        return avg;
    },0);
};

File: leetcode/algorithms/array/average-salary-excluding-the-minimum-and-maximum-salary.java
/*
    leetcode
    category: "Algorithms"
    problem title: "Average Salary Excluding the Minimum and Maximum Salary"
*/

class Solution {
    public double average(int[] salary) {
        
        int max=salary[0],min=salary[0],sum=0;
        for(int sal : salary){
            sum+=sal;
            if(sal>max){
                max=sal;
            }
            if(sal<min){
                min=sal;
            }
        }
        
        
        return (double)((sum-(max+min)))/(salary.length-2);
        
    }
}